### PR TITLE
Refactor fully connected and change to explicit test

### PR DESCRIPTION
**Pull request by Jopyth**
We encapsulate that the layer is storing weights transposed.

### CODE REVIEW

File: length/tests/layer_tests/test_fully_connected.py [hunk 1]
```diff
@@ -23,7 +55,7 @@ def test_fully_connected_backward():
     comp_grad_x, comp_grad_weight, comp_grad_bias = layer.internal_backward((data,), (gradient,))
 
     f = lambda: layer.internal_forward((data,))
-    num_grad_x, num_grad_weight, num_grad_bias = gradient_checker.compute_numerical_gradient(f, (data, layer.weights, layer.bias), (gradient,), eps=1e-2)
+    num_grad_x, num_grad_weight, num_grad_bias = gradient_checker.compute_numerical_gradient(f, (data, layer.weights.T, layer.bias), (gradient,), eps=1e-2)
```
Comment (Jopyth) [inline length/tests/layer_tests/test_fully_connected.py]: Something does not seem right here.

File: length/layers/fully_connected.py [hunk 2]
```diff
@@ -13,28 +13,44 @@ class FullyConnected(Layer):
...
+
+    @property
+    def weights(self):
+        return self._weights.T
+
+    @weights.setter
+    def weights(self, value):
+        self._weights = value.T
+
+    @property
```
Comment (Bartzi) [inline length/layers/fully_connected.py]: I'd say there is no need for a `bias` property function, as we are not doing anything there, right?

File: length/tests/layer_tests/test_fully_connected.py [hunk 3]
```diff
@@ -6,12 +6,44 @@
 from length.tests import gradient_checker
 
 
+def init(array):
+    return np.array(array).astype(constants.DTYPE)
```
Comment (Bartzi) [inline length/tests/layer_tests/test_fully_connected.py]: you could also write `np.array(array, dtype=constants.DTYPE)`

File: length/tests/layer_tests/test_fully_connected.py (hunk 1 above)
Comment (Bartzi) [reply length/tests/layer_tests/test_fully_connected.py]: why no `layer._weights`?

**Review by Bartzi — COMMENTED** [review]
:boom:

File: length/tests/layer_tests/test_fully_connected.py (hunk 3 above)
Comment (Jopyth) [reply length/tests/layer_tests/test_fully_connected.py]: :man_facepalming:

File: length/layers/fully_connected.py (hunk 2 above)
Comment (Jopyth) [reply length/layers/fully_connected.py]: maybe someone else wants to?

**Review by Bartzi — APPROVED** [review]
:small_airplane:

**Pull request merged**